hugr-core/src/hugr/views/sibling_subgraph.rs

```diff
@@ -649,8 +689,21 @@ pub enum InvalidSubgraph {
     #[error("The subgraph is not convex.")]
     NotConvex,
     /// Not all nodes have the same parent.
-    #[error("Not a sibling subgraph.")]
-    NoSharedParent,
+    #[error(
+        "Invalid sibling subgraph. {first_node} has parent {}, but {other_node} has parent {}.",
```

Comment:
I quite like "not a" sibling subgraph (the definition of such is that they are all siblings!), but ok